lib/wovnrb.rb

```diff
@@ -29,14 +29,21 @@ def call(env)
 
       @env = env
       headers = Headers.new(env, @store.settings)
+      default_lang = @store.settings['default_lang']
       return @app.call(env) if @store.settings['test_mode'] && @store.settings['test_url'] != headers.url
 
       # redirect if the path is set to the default language (for SEO purposes)
-      if headers.path_lang == @store.settings['default_lang']
-        redirect_headers = headers.redirect(@store.settings['default_lang'])
+      if headers.path_lang == default_lang
+        redirect_headers = headers.redirect(default_lang)
         return [307, redirect_headers, ['']]
       end
 
+      # if path containing language code is ignored, do nothing
+      if headers.lang_code != default_lang && ignore_path?(headers.unmasked_pathname_without_trailing_slash)
```

Comment:
Because the case here is to do nothing when language is target and dir with target language is ignored. Not checking the original language will change behavior outside of current user need and will affect all users that have ignore paths (after they update).

Reply:
The only extra code that seems like it will be executed in the default language case is
`status, res_headers, body = @app.call(headers.request_out)` and `@store.settings.update_dynamic_settings!(request.params)`.  What's the difference between `output` and `@app.call(headers.request_out)`?

Reply:
It is not about difference between `output` and `@app.call(headers.request_out)` (because there is none) but about code that could be executed afterwards. I do think it is fine to not check for original language here, but I feel it is safer.